src/track/track.h

```diff
@@ -217,6 +216,26 @@ class Track : public QObject {
     void setTrackNumber(const QString&);
     void setTrackTotal(const QString&);
 
+    /// Return the genre as text
+    QString getGenre() const;
+    /// !!!DO NOT USE!!!
+    /// Set the genre text WITHOUT updating the corresponding custom tags.
+    ///
+    /// Only allowed to be used by TrackDAO!!! Unfortunately, the
+    /// design of TrackDAO does not allow to hide this method by
+    /// making it private.
+    void setGenreInternal(
+            const QString& genre);
+    bool updateGenre(
+            const QString& genre);
```

Comment:
The comment doesn't make it clear to me if it only applies to `setGenreInternal` or to `updateGenre` as well. Maybe add a blank line between them and a separate doc comment.

Should we make the interal method private/protected and make TrackDAO a friend class to prevent wrong usage?